tests/unittests/memoryAllocatorTest.cpp

```diff
@@ -111,3 +111,17 @@ TEST(MemAlloc, allocateToTheMax) {
 
   EXPECT_EQ(MA.getMaxMemoryUsage(), 128);
 }
+
+/// Since MemoryAllocator is used for different purposes, including the
+/// modelling of a memory allocation process on a target device, the types used
+/// to represent the allocated addresses should be unsigned and at least 64-bit
+/// wide.
+TEST(MemAlloc, testMemoryAllocatorTypes) {
```

Comment:
I think that this unit test can be replaced with a static assert.

Reply:
Good idea! Added a static_assert instead of this unit test.